From fd4e430b7ae2b52c44f6764905ede872bfc9ac1f Mon Sep 17 00:00:00 2001
From: Ultrabug <ultrabug@gentoo.org>
Date: Thu, 4 May 2017 17:31:21 +0200
Subject: [PATCH] Add support for Gentoo Linux irqbalance configuration
 detection.

This allows perftune.py to work well when using the irqbalance
daemon on Gentoo Linux.

Signed-off-by: Ultrabug <ultrabug@gentoo.org>
---
 seastar/scripts/perftune.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/seastar/scripts/perftune.py b/seastar/scripts/perftune.py
index 6748086..1b88d8c 100755
--- a/seastar/scripts/perftune.py
+++ b/seastar/scripts/perftune.py
@@ -86,6 +86,11 @@ def restart_irqbalance(banned_irqs):
             config_file = '/etc/sysconfig/irqbalance'
             options_key = 'IRQBALANCE_ARGS'
             systemd = True
+        elif os.path.exists('/etc/conf.d/irqbalance'):
+            config_file = '/etc/conf.d/irqbalance'
+            options_key = 'IRQBALANCE_OPTS'
+            with open('/proc/1/comm', 'r') as comm:
+                systemd = 'systemd' in comm.read()
         else:
             print("Unknown system configuration - not restarting irqbalance!")
             print("You have to prevent it from moving IRQs {} manually!".format(banned_irqs_list))
-- 
2.12.2

